Actron Air: Add switch entity platform

kclif9 opened the pull request:
## Proposed change
This PR adds the Switch entity platform to the Actron Air integration, supporting the features Away Mode, Continuous Fan, Quiet Mode and Turbo Mode. 


## Type of change
- [ ] Dependency upgrade
- [ ] Bugfix (non-breaking change which fixes an issue)
- [ ] New integration (thank you!)
- [X] New feature (which adds functionality to an existing integration)
- [ ] Deprecation (breaking change to happen in the future)
- [ ] Breaking change (fix/feature causing existing functionality to break)
- [ ] Code quality improvements to existing code or addition of tests

## Additional information
- This PR fixes or closes issue: fixes #
- This PR is related to issue: 
- Link to documentation pull request: https://github.com/home-assistant/home-assistant.io/pull/42343
- Link to developer documentation pull request: 
- Link to frontend pull request: 

## Checklist
- [X] I understand the code I am submitting and can explain how it works.
- [X] The code change is tested and works locally.
- [X] Local tests pass. **Your PR cannot be merged unless tests pass**
- [X] There is no commented out code in this PR.
- [X] I have followed the [development checklist][dev-checklist]
- [X] I have followed the [perfect PR recommendations][perfect-pr]
- [X] The code has been formatted using Ruff (`ruff format homeassistant tests`)
- [X] Tests have been added to verify that the new code works.
- [X] Any generated code has been carefully reviewed for correctness and compliance with project standards.

If user exposed functionality or configuration variables are added/changed:

- [X] Documentation added/updated for [www.home-assistant.io][docs-repository]

If the code communicates with devices, web services, or third-party tools:

- [X] The [manifest file][manifest-docs] has all fields filled out correctly.  
      Updated and included derived files by running: `python3 -m script.hassfest`.
- [X] New or updated dependencies have been added to `requirements_all.txt`.  
      Updated by running `python3 -m script.gen_requirements_all`.
- [X] For the updated dependencies - a link to the changelog, or at minimum a diff between library versions is added to the PR description.

<!--
  This project is very active and we have a high turnover of pull requests.

  Unfortunately, the number of incoming pull requests is higher than what our
  reviewers can review and merge so there is a long backlog of pull requests
  waiting for review. You can help here!
  
  By reviewing another pull request, you will help raise the code quality of
  that pull request and the final review will be faster. This way the general
  pace of pull request reviews will go up and your wait time will go down.
  
  When picking a pull request to review, try to choose one that hasn't yet
  been reviewed.

  Thanks for helping out!
-->

To help with the load of incoming pull requests:

- [ ] I have reviewed two other [open pull requests][prs] in this repository.

[prs]: https://github.com/home-assistant/core/pulls?q=is%3Aopen+is%3Apr+-author%3A%40me+-draft%3Atrue+-label%3Awaiting-for-upstream+sort%3Acreated-desc+review%3Anone+-status%3Afailure

<!--
  Thank you for contributing <3

  Below, some useful links you could explore:
-->
[dev-checklist]: https://developers.home-assistant.io/docs/development_checklist/
[manifest-docs]: https://developers.home-assistant.io/docs/creating_integration_manifest/
[quality-scale]: https://developers.home-assistant.io/docs/integration_quality_scale_index/
[docs-repository]: https://github.com/home-assistant/home-assistant.io
[perfect-pr]: https://developers.home-assistant.io/docs/review-process/#creating-the-perfect-pr

kclif9 committed:
Add switch entity platform

kclif9 committed:
Merge branch 'dev' into Actron-Air-Add-Switch-Entity-Type

kclif9 committed:
fix copilot suggestions

NoRi2909 commented on homeassistant/components/actron_air/strings.json:
```diff
@@ -25,5 +25,21 @@
+        "name": "Quiet Mode"
+      },
+      "turbo_mode": {
+        "name": "Turbo Mode"
```
These should all be sentence-cased to match Home Assistant guidelines:

```suggestion
      "away_mode": {
        "name": "Away mode"
      },
      "continuous_fan": {
        "name": "Continuous fan"
      },
      "quiet_mode": {
        "name": "Quiet mode"
      },
      "turbo_mode": {
        "name": "Turbo mode"
```

joostlek commented on homeassistant/components/actron_air/switch.py:
```diff
@@ -0,0 +1,155 @@
+
+    _attr_has_entity_name = True
+    _attr_entity_category = EntityCategory.CONFIG
+    _attr_device_class = SwitchDeviceClass.SWITCH
```
I beliefe this is more for a light switch

kclif9 committed:
Refactor Actron Air switch platform and update tests

- Standardize switch names in strings.json to lowercase.
- Introduce ActronAirSwitchEntityDescription for better switch entity management.
- Simplify switch setup logic in async_setup_entry.
- Enhance test coverage for switch entities and their functionalities.
- Mock Actron Air API responses for comprehensive testing.

kclif9 committed:
Merge branch 'dev' into Actron-Air-Add-Switch-Entity-Type

kclif9 committed:
Use snapshot test per PR comment

kclif9 committed:
remove line based on PR comment

kclif9 committed:
Merge branch 'dev' into Actron-Air-Add-Switch-Entity-Type

joostlek reviewed (changes requested):
Tests are failing and there's a merge conflict

kclif9 committed:
Merge branch 'dev' into Actron-Air-Add-Switch-Entity-Type

kclif9 committed:
setting unique_id to _ to be consistent with entity id naming standard

kclif9 committed:
Merge branch 'dev' into Actron-Air-Add-Switch-Entity-Type

joostlek committed:
Fix
